formatTest/typeCheckedTests/expected_output/reasonComments.re

```diff
@@ -61,36 +73,31 @@ let testingEndOfLineComments = [
   "Item 3", /* Comment For Third Item */
   "Item 4" /* Comment For Fourth Item - but before trailing comma */
   /* Comment after last item in list. */
-]
-/* Comment after rbracket */;
+] /* Comment after rbracket */;
 
 /* The space between ; and comment shoudn't matter */
 let testPlacementOfTrailingComment = [
   "Item 0" /* */
   /* Comment after last item in list. */
-];
+]; /* Comment after semi */
```

Comment:
I would suggest doing that (formatting to the top of the entire statement). It's useful to have a distinction between perfectly end-of-line comments, and imperfect ones like this.
Imperfect:
```rust
let x = (
   "first",
   "second"
) /* imperfect eol comment on tuple */
;
```
Perfect
```rust
let x =
  ("first", "second") /* perfect eol comment on tuple */
;
```


Maybe this is another category of comment called "PartialEolComment" which would be normalized to be above the item when printing. In my previous attempts, I made the mistake of not making the distinction between imperfect and perfect eol comments. That causes much difficulty.

```rust
type x =
   Variant (
   really *
   long *
   tupleType
   ...
) /* Comment */
;
```

In that example, it's an imperfect eol comment is on the tuple portion of the variant type. If we just treated it as a regular eol comment, then we have a problem, because the only way to ensure that eol comments remain as eol comments across different width formattings, is to force `~break:Never` on the entire AST item (and force break on the containing list/label). That's a problem because these imperfect eol comments are typically on very very long structures, so it will cause massively long unbroken line. The solution seems to be to consider these as different "imperfect eol comments", and to format the comment above the AST node instead.